docs/extensions/tutorials/_posts/1970-01-05-Installing3rdPartyPackages.md

```diff
@@ -0,0 +1,323 @@
+  "dependencies": {
+    "react-native-camera": "^0.4.1"
+  },
+  "rnpm": {
+    "commands": {
+      "postlink": "node node_modules/{{ site.example.devName }}.qr-reader-extension/scripts/run.js"
+    }
+  }
+}
+```
+
+We have to make a `scripts` directory and make the postlink script in it: `app/scripts/run.js`
+
+```ShellSession
+$ cd app
+$ mkdir scripts
+$ cd scripts
+$ touch run.js
+```
+
+It's going to edit the `Info.plist` file and link the react-native-camera dependency for our app.
+
+```javascript{1-21}
+#file: app/scripts/run.js
+const fs = require('fs-extra');
+const plist = require('plist');
+
+const infoPlistPath = './ios/ShoutemApp/Info.plist';
+const infoPlistFile = fs.readFileSync(infoPlistPath, 'utf8');
+const infoPlist = plist.parse(infoPlistFile);
+
+console.log('Adding camera and microphone permissions to Info.plist');
+infoPlist['NSCameraUsageDescription'] = 'App needs your camera to be able to scan QR codes';
+infoPlist['NSMicrophoneUsageDescription'] = 'App needs your microphone to be able to scan QR codes';
+fs.writeFileSync(infoPlistPath, plist.build(infoPlist));
+
+const exec = require('child_process').execSync;
+
+const dependenciesToLink = ['react-native-camera'];
+
+const command = 'node node_modules/react-native/local-cli/cli.js link';
+
+dependenciesToLink.forEach((dependency) => {
+  exec(`${command} ${dependency}`);
+});
+```
+
+Now let's explain the details regarding native dependency linking in `app/scripts/run.js`
+  - everything regarding `Info.plist` is added to notify the user that the app is using his camera
+  - `dependenciesToLink` is an array that stores all of our native dependencies
+  - `command` represents the react-native-cli, used to link native dependencies
+  - the final three lines of code make sure each dependency in our array is linked
+
+The reason we create an array is because sometimes our extension will have multiple native dependencies, so instead of making a separate linking command for each, we simply run one `forEach` loop and hand it an array of dependencies.
+
+### Using the Package
+
+Our extension will now have access to everything react-native-camera has to offer. For this example we'll edit `app/screens/QRReaderScreen.js` so that it displays an alert when the camera reads a QR code and the alert message will contain the QR code data.
+
+```javascript{2-4,6-10,12-21,25-28}
+#file: app/screens/QRReaderScreen.js
+import React, { Component } from 'react';
+import { Alert } from 'react-native';
+import Camera from 'react-native-camera';
+import _ from 'lodash';
+
+export default class QRReaderScreen extends Component {
+    constructor(props) {
+      super(props);
+      this.onBarCodeRead = this.onBarCodeRead.bind(this);
+    }
+
+  onBarCodeRead(code) {
+      Alert.alert(
+        'QR Code Detected',
+        code.data,
+        [
+          {text: 'OK, read it.', onPress: () => console.log('User saw QR Code contents.')},
+        ],
+        { cancelable: false }
```

Comment:
Once you set it up, go through examples you wrote and update extension-examples.